gt/main.c

```diff
@@ -93,24 +96,36 @@ gt_parse_incoming_pkt(struct rte_mbuf *pkt, struct gt_packet_headers *info)
 			return -1;
 
 		outer_ipv4_hdr = (struct ipv4_hdr *)info->outer_l3_hdr;
-		parsed_len += sizeof(struct ipv4_hdr);
+		outer_ipv4_hdr_len = ipv4_hdr_len(outer_ipv4_hdr);
+		parsed_len += outer_ipv4_hdr_len;
 		info->priority = (outer_ipv4_hdr->type_of_service >> 2);
 		info->outer_ecn =
 			outer_ipv4_hdr->type_of_service & IPTOS_ECN_MASK;
 		encasulated_proto = outer_ipv4_hdr->next_proto_id;
 		break;
-	case ETHER_TYPE_IPv6:
+	case ETHER_TYPE_IPv6: {
+		uint8_t nexthdr;
```

Comment:
Once we drop variable `nexthdr` (see next comment), we don't need the braces for this case.